jbi/steps.py

```diff
@@ -200,6 +201,56 @@ def maybe_assign_jira_user(
     return (StepStatus.NOOP, context)
 
 
+def maybe_update_issue_milestone(
+    context: ActionContext, *, parameters: ActionParams, jira_service: JiraService
+) -> StepResult:
+    """
+    Update the Jira issue 'Fix Version' field.
+    """
+    # Do not bother update the field if empty on create or if another field is updated.
+    if (context.operation == Operation.CREATE and not context.bug.milestone) or (
+        context.operation == Operation.UPDATE
+        and "milestone" not in context.event.changed_fields()
+    ):
+        return (StepStatus.NOOP, context)
+
+    if context.bug.milestone:
+        m = re.match(r"Branch (\d+)", context.bug.milestone)
+        if m is None:
+            # Milestone not supported (eg. "Future")
+            return (StepStatus.NOOP, context)
+
+        milestone_version = m.group(1)
+        jira_project_versions = {
+            v["name"]
+            for v in jira_service.client.get_project_versions(context.jira.project)
+        }
+        potential_fix_versions = {
+            f"Fx{milestone_version}",
+            f"Release {milestone_version}",
+        }
+        jira_fix_versions = list(
+            jira_project_versions.intersection(potential_fix_versions)
+        )
+        if not jira_fix_versions:
+            logger.info(
+                "Bug %s milestone %r has no matching version in project.",
```

Comment:
Where will the "Bug %s milestone %r has no matching version in project." errors be logged when a Bugzilla Milestone name doesn't match the Jira Release? Is that information only available to JBI admins?

Reply:
These are application/server logs that end-up in Google Cloud console. We currently don't have any way to send alerts to users (we could consider adding a comment on the bug or on the Jira issue)